docs/execution-providers/TensorRT-ExecutionProvider.md

```diff
@@ -183,8 +189,8 @@ session_options.AppendExecutionProvider_TensorRT(trt_options);
 #### Python API example
 ```
 sess_opt = ort.SessionOptions()
-sess = ort.InferenceSession('model.onnx', sess_options=sess_opt)
-sess.set_providers(["TensorrtExecutionProvider"],[{'device_id': '1', 'trt_max_workspace_size': '2147483648', 'trt_fp16_enable':'True'}])
+sess = ort.InferenceSession('model.onnx', sess_options=sess_opt, providers=['TensorrtExecutionProvider'],
+                            provider_options=[{'device_id': '1', 'trt_max_workspace_size': '2147483648', 'trt_fp16_enable':'True'}])
```

Comment:
We can't pass `providers` and `provider_options` at the same time to the InferenceSession ctor. To pass both, they need to be merged together and sent in as `providers`

Reply:
Found this out while I was doing something similar in https://github.com/microsoft/onnxruntime/pull/9916/

Reply:
thanks for the heads up. that seems contrary to what's documented or what's intuitive. 
i wonder if it previously worked but was broken once we started accepting tuples for provider + options dict

Reply:
i can change the example to use tuple for now.

Reply:
I agree that it is unintuitive the way it is currently. I am not sure if it was working previously and if it stopped working after the change you are referring to. It took me some time to figure this out today morning while I was doing the other doc update. So, I thought I will save you the time.

Reply:
> 

thanks!

Reply:
Based on this test: https://github.com/microsoft/onnxruntime/blob/e61301917442486e6ef9c7f39cddab72739f0fc5/onnxruntime/test/python/onnxruntime_test_python.py#L1062, there shouldn't be anything wrong with `check_and_normalize_provider_args()`. I wonder if there is some other logic in `_create_inference_session()` in InferenceSession ctor that is causing a problem. One would have to debug this as I couldn't get it working without the tuple approach.

Reply:
+@chilo-ms , can you help look into why provider_options doesn't seem to be working when passed into InferenceSession constructor? you had originally added the python provider_options support, so maybe something has regressed since then.